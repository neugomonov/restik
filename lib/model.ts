import { GLTFLoader } from "three/examples/jsm/loaders/GLTFLoader.js";

export function loadGLTFModel(
	scene: any,
	glbPath: any,
	options = { receiveShadow: true, castShadow: true }
) {
	const { receiveShadow, castShadow } = options;
	return new Promise((resolve, reject) => {
		const loader = new GLTFLoader();

		loader.load(
			glbPath,
			(gltf: any) => {
				const obj = gltf.scene;
				obj.name = "pizza";
				obj.position.y = 0;
				obj.position.x = 0;
				obj.receiveShadow = receiveShadow;
				obj.castShadow = castShadow;
				scene.add(obj);

				obj.traverse(function (child: any) {
					if (child.isMesh) {
						child.castShadow = castShadow;
						child.receiveShadow = receiveShadow;
					}
				});
				resolve(obj);
			},
			undefined,
			function (error: any) {
				reject(error);
			}
		);
	});
}
