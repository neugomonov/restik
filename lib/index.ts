const index = {
	videoCapture: {
		en: "How do we make our pizza so delicious?",
		ru: "Как мы делаем нашу пиццу такой вкусной",
	},
	chat: {
		en: "Ask questions in the chat with Pizzeria, and we'll answer right away!",
		ru: "Задавайте вопросы в чат с Пиццерией, и мы обязательно ответим!",
	},
};

export default index;
