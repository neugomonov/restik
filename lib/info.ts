const info = {
	name: "Пиццерия",
	title: {
		en: "Pizzeria",
		ru: "Пиццерия",
	},
	description: {
		en: "Your favorite pizzeria! We offer fresh and tasty food and low prices.",
		ru: "Ваша любимая пиццерия! Мы предлагаем свежую и вкусную еду и низкие цены.",
	},
	notifications: {
		en: "Notifications",
		ru: "Оповещения",
	},
	translate: {
		en: "Translate",
		ru: "Перевод",
	},
	menu: {
		en: "Menu",
		ru: "Меню",
	},
	video: {
		en: "Video",
		ru: "Видео",
	},
	news: {
		en: "News",
		ru: "Новости",
	},
	promo: {
		en: "Promo",
		ru: "Акции",
	},
	currency: "₽",
	callingCode: "+7",
	schedule: {
		0: {
			opens: "12:00",
			closes: "22:00",
		},
		1: {
			opens: "12:00",
			closes: "22:00",
		},
		2: {
			opens: "12:00",
			closes: "22:00",
		},
		3: {
			opens: "12:00",
			closes: "22:00",
		},
		4: {
			opens: "12:00",
			closes: "22:00",
		},
		5: {
			opens: "12:00",
			closes: "23:00",
		},
		6: {
			opens: "0",
			closes: "0",
		},
	},
	averageDelivery: 1,
	holidays: ["17/09"],
	types: {
		small: {
			en: "Small",
			ru: "Немного",
		},
		large: {
			en: "Large",
			ru: "Много",
		},
	},
	isDevelopment: true,
};

export default info;
