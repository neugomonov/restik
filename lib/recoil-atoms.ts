import { atom } from "recoil";

export interface CartState {
	items: Array<{ name: string; type: string; price: number; quantity: number }>;
	total: number;
}

// Cart
export const _cart = atom<CartState>({
	key: "theme",
	default: {
		items: [],
		total: 0,
	},
});
