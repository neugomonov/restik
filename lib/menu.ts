import info from "./info";

const { types } = info;

const translatedIngredients = new Map([
	["Margherita", { en: "Margherita", ru: "Маргарита" }],
	["Capricciosa", { en: "Capricciosa", ru: "Капричоза" }],
	["Napoletana", { en: "Napoletana", ru: "Наполетана" }],
	["Calzone", { en: "Calzone", ru: "Кальцоне" }],
	["Quattro-Formaggi", { en: "Quattro-Formaggi", ru: "Четыре сыра" }],
	["Caprese", { en: "Caprese", ru: "Капрезе" }],
	["Tarhun", { en: "Tarhun", ru: "Тархун" }],
	["Pomegranate-juice", { en: "Pomegranate juice", ru: "Гранатовый сок" }],
	["Latte", { en: "Latte", ru: "Латте" }],

	["Cheese", { en: "Cheese", ru: "Сыр" }],
	["Mushrooms", { en: "Mushrooms", ru: "Грибы" }],
	["Ham", { en: "Ham", ru: "Ветчина" }],
	["Olives", { en: "Olives", ru: "Оливки" }],
	["Oregano", { en: "Oregano", ru: "Душица " }],
	["Pepper", { en: "Pepper", ru: "Перец" }],
	["Rice", { en: "Rice", ru: "Рис" }],
	["Saffron", { en: "Saffron", ru: "Шафран" }],
	["Turmeric", { en: "Turmeric", ru: "Куркума" }],
	["Chicken", { en: "Chicken", ru: "Курица" }],
]);

const menu = (lang: "en" | "ru") => [
	{
		name: (
			translatedIngredients.get("Margherita") as { en: string; ru: string }
		)[lang],
		image: "images/covers/margherita.webp",
		ingredients: [
			(translatedIngredients.get("Chicken") as { en: string; ru: string })[
				lang
			],
			(translatedIngredients.get("Oregano") as { en: string; ru: string })[
				lang
			],
		],
		variants: [
			{ type: types.small[lang], price: 280 },
			{ type: types.large[lang], price: 380 },
		],
	},
	{
		name: (
			translatedIngredients.get("Capricciosa") as { en: string; ru: string }
		)[lang],
		image: "images/covers/capricciosa.jpg",
		ingredients: [
			(translatedIngredients.get("Chicken") as { en: string; ru: string })[
				lang
			],
			(translatedIngredients.get("Pepper") as { en: string; ru: string })[lang],
			(translatedIngredients.get("Oregano") as { en: string; ru: string })[
				lang
			],
		],
		variants: [
			{ type: types.small[lang], price: 120 },
			{ type: types.large[lang], price: 200 },
		],
	},
	{
		name: (
			translatedIngredients.get("Napoletana") as { en: string; ru: string }
		)[lang],
		image: "images/covers/napoletana.jpg",
		ingredients: [
			(translatedIngredients.get("Rice") as { en: string; ru: string })[lang],
			(translatedIngredients.get("Saffron") as { en: string; ru: string })[
				lang
			],
			(translatedIngredients.get("Turmeric") as { en: string; ru: string })[
				lang
			],
			(translatedIngredients.get("Chicken") as { en: string; ru: string })[
				lang
			],
		],
		variants: [
			{ type: types.small[lang], price: 210 },
			{ type: types.large[lang], price: 390 },
		],
	},
	{
		name: (translatedIngredients.get("Calzone") as { en: string; ru: string })[
			lang
		],
		image: "images/covers/Calzone.jpg",
		ingredients: [
			(translatedIngredients.get("Chicken") as { en: string; ru: string })[
				lang
			],
			(translatedIngredients.get("Mushrooms") as { en: string; ru: string })[
				lang
			],
			(translatedIngredients.get("Ham") as { en: string; ru: string })[lang],
			(translatedIngredients.get("Olives") as { en: string; ru: string })[lang],
			(translatedIngredients.get("Cheese") as { en: string; ru: string })[lang],
		],
		variants: [
			{ type: types.small[lang], price: 200 },
			{ type: types.large[lang], price: 380 },
		],
	},
	{
		name: (
			translatedIngredients.get("Quattro-Formaggi") as {
				en: string;
				ru: string;
			}
		)[lang],
		image: "images/covers/Quattro-Formaggi.jpg",
		ingredients: [
			(translatedIngredients.get("Chicken") as { en: string; ru: string })[
				lang
			],
			(translatedIngredients.get("Mushrooms") as { en: string; ru: string })[
				lang
			],
			(translatedIngredients.get("Ham") as { en: string; ru: string })[lang],
			(translatedIngredients.get("Olives") as { en: string; ru: string })[lang],
		],
		variants: [
			{ type: types.small[lang], price: 210 },
			{ type: types.large[lang], price: 390 },
		],
	},
	{
		name: (translatedIngredients.get("Caprese") as { en: string; ru: string })[
			lang
		],
		image: "images/covers/caprese.jpg",
		ingredients: [
			(translatedIngredients.get("Chicken") as { en: string; ru: string })[
				lang
			],
			(translatedIngredients.get("Mushrooms") as { en: string; ru: string })[
				lang
			],
			(translatedIngredients.get("Ham") as { en: string; ru: string })[lang],
			(translatedIngredients.get("Olives") as { en: string; ru: string })[lang],
		],
		variants: [
			{ type: types.small[lang], price: 150 },
			{ type: types.large[lang], price: 270 },
		],
	},
	{
		name: (translatedIngredients.get("Tarhun") as { en: string; ru: string })[
			lang
		],
		image: "images/covers/tarhun.jpg",
		ingredients: [],
		variants: [
			{ type: types.small[lang], price: 60 },
			{ type: types.large[lang], price: 100 },
		],
	},
	{
		name: (
			translatedIngredients.get("Pomegranate-juice") as {
				en: string;
				ru: string;
			}
		)[lang],
		image: "images/covers/pomegranate-juice.jpg",
		ingredients: [],
		variants: [
			{ type: types.small[lang], price: 70 },
			{ type: types.large[lang], price: 120 },
		],
	},
	{
		name: (translatedIngredients.get("Latte") as { en: string; ru: string })[
			lang
		],
		image: "images/covers/Latte.jpg",
		ingredients: [],
		variants: [
			{ type: types.small[lang], price: 150 },
			{ type: types.large[lang], price: 250 },
		],
	},
];

export default menu;
