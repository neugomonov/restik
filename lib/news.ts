import info from "./info";

const { types } = info;
const translatedIngredients = new Map([
	[
		"Courier",
		{
			en: "Track the courier in our application!",
			ru: "Отследи курьера в нашем приложении!",
		},
	],
	[
		"Year",
		{
			en: "Winter offers 2022 in Pizzeria!",
			ru: "Зимние предложения 2022 в Пиццерии!",
		},
	],
	[
		"Geo",
		{
			en: "You can order delivery by geolocation!",
			ru: "Вы можете заказать доставку по геопозиции! ",
		},
	],
	[
		"Pay",
		{
			en: "How to pay for an order at Pizzeria?",
			ru: "Как оплатить заказ в Пиццерии?",
		},
	],
	[
		"Lottery",
		{
			en: "Results of the big New Year's draw!",
			ru: "Итоги большого новогоднего розыгрыша!",
		},
	],
	[
		"Code",
		{
			en: "Pizzeria conducts IT CODE 2.0!",
			ru: "Пиццерия проводит IT CODE 2.0!",
		},
	],
	["Logo", { en: "Logo Contest", ru: "Конкурс на  логотип" }],
	["LotteryEnd", { en: "Lottery results", ru: "Итоги лотереи" }],
	["Opened", { en: "We have opened! 🎉", ru: "Мы открылись! 🎉" }],

	[
		"ToBeContinued",
		{
			en: "Order home, eat with us, follow our news, there will be a lot of interesting things...",
			ru: "Заказывайте домой, ешьте у нас, следите за нашими новостями, будет много интересного...",
		},
	],
]);

const news = (lang: "en" | "ru") => [
	{
		name: (translatedIngredients.get("Courier") as { en: string; ru: string })[
			lang
		],
		image: "images/covers/news/live.jpg",
		ingredients: [
			(
				translatedIngredients.get("ToBeContinued") as { en: string; ru: string }
			)[lang],
		],
		date: "10.04.2022",
		variants: [
			{ type: types.small[lang], price: 280 },
			{ type: types.large[lang], price: 380 },
		],
	},
	{
		name: (translatedIngredients.get("Pay") as { en: string; ru: string })[
			lang
		],
		image: "images/covers/news/howsway.jpg",
		ingredients: [
			(
				translatedIngredients.get("ToBeContinued") as { en: string; ru: string }
			)[lang],
		],
		date: "08.04.2022",
		variants: [
			{ type: types.small[lang], price: 200 },
			{ type: types.large[lang], price: 380 },
		],
	},
	{
		name: (
			translatedIngredients.get("Lottery") as {
				en: string;
				ru: string;
			}
		)[lang],
		image: "images/covers/news/500.jpg",
		ingredients: [
			(
				translatedIngredients.get("ToBeContinued") as { en: string; ru: string }
			)[lang],
		],
		date: "08.04.2022",
		variants: [
			{ type: types.small[lang], price: 210 },
			{ type: types.large[lang], price: 390 },
		],
	},
	{
		name: (translatedIngredients.get("Year") as { en: string; ru: string })[
			lang
		],
		image: "images/covers/news/newyear.jpg",
		ingredients: [
			(
				translatedIngredients.get("ToBeContinued") as { en: string; ru: string }
			)[lang],
		],
		date: "07.04.2022",
		variants: [
			{ type: types.small[lang], price: 120 },
			{ type: types.large[lang], price: 200 },
		],
	},
	{
		name: (translatedIngredients.get("Code") as { en: string; ru: string })[
			lang
		],
		image: "images/covers/news/itcode.jpg",
		ingredients: [
			(
				translatedIngredients.get("ToBeContinued") as { en: string; ru: string }
			)[lang],
		],
		date: "07.04.2022",
		variants: [
			{ type: types.small[lang], price: 150 },
			{ type: types.large[lang], price: 270 },
		],
	},
	{
		name: (translatedIngredients.get("Geo") as { en: string; ru: string })[
			lang
		],
		image: "images/covers/news/doyou.jpg",
		ingredients: [
			(
				translatedIngredients.get("ToBeContinued") as { en: string; ru: string }
			)[lang],
		],
		date: "05.04.2022",
		variants: [
			{ type: types.small[lang], price: 210 },
			{ type: types.large[lang], price: 390 },
		],
	},
	{
		name: (translatedIngredients.get("Logo") as { en: string; ru: string })[
			lang
		],
		image: "images/covers/news/logo.jpg",
		ingredients: [
			(
				translatedIngredients.get("ToBeContinued") as { en: string; ru: string }
			)[lang],
		],
		date: "04.04.2022",
		variants: [
			{ type: types.small[lang], price: 60 },
			{ type: types.large[lang], price: 100 },
		],
	},
	{
		name: (
			translatedIngredients.get("LotteryEnd") as {
				en: string;
				ru: string;
			}
		)[lang],
		image: "images/covers/news/10.jpg",
		ingredients: [
			(
				translatedIngredients.get("ToBeContinued") as { en: string; ru: string }
			)[lang],
		],
		date: "03.04.2022",
		variants: [
			{ type: types.small[lang], price: 70 },
			{ type: types.large[lang], price: 120 },
		],
	},
	{
		name: (translatedIngredients.get("Opened") as { en: string; ru: string })[
			lang
		],
		image: "images/covers/news/openup.jpg",
		ingredients: [
			(
				translatedIngredients.get("ToBeContinued") as { en: string; ru: string }
			)[lang],
		],
		date: "03.04.2022",
		variants: [
			{ type: types.small[lang], price: 150 },
			{ type: types.large[lang], price: 250 },
		],
	},
];

export default news;
