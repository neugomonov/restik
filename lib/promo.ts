import info from "./info";

const { types } = info;
const translatedIngredients = new Map([
	[
		"OPENUP",
		{
			en: "Pizza as a gift for the whole company!",
			ru: "Пицца в подарок для всей компании!",
		},
	],
	[
		"PIZZA1999",
		{
			en: "Come to Pizzeria for a profit!",
			ru: "Приходите за выгодой в Пиццерию!",
		},
	],
	[
		"Prize",
		{
			en: "We give pizza for the first order!",
			ru: "Дарим пиццу за первый заказ! ",
		},
	],
	["Souvenir", { en: "Souvenir as a gift!", ru: "Сувенир в подарок!" }],
	["Invite", { en: "Get a friend!", ru: "Пригласите друга!" }],
	["Birthday", { en: "Birthday pizza!", ru: "Пицца на день рождения!" }],
	[
		"FirstOrder",
		{ en: "2 pizzas for the first order!", ru: "2 пиццы за первый заказ!" },
	],
	["FIRST1", { en: "30% discount at Pizzeria!", ru: "Скидка 30% в Пиццерии!" }],
	[
		"MakeSet",
		{ en: "Order a set for only 795 ₽", ru: "Закажите сет всего за 795 ₽" },
	],

	[
		"1999",
		{
			en: "For an order from 1999 ₽ a discount of 400 ₽ using the promo code PIZZA1999! 🍕",
			ru: "За заказ от 1999 рублей скидка 400 ₽ по промокоду PIZZA1999! 🍕",
		},
	],
	[
		"30",
		{
			en: "Get 30% off your first order with promo code FIRST1! 1️⃣",
			ru: "За первый заказ скидка 30% по промокоду FIRST1! 1️⃣",
		},
	],
	[
		"10",
		{
			en: "We have opened! 🥳 10% off everything with promo code OPENUP!",
			ru: "Мы открылись! 🥳 Скидка 10% на всё с промокодом OPENUP!",
		},
	],
	[
		"Tasty",
		{
			en: "For delivery orders from 999 ₽ we give delicious gifts!",
			ru: "За заказ доставки от 999 рублей дарим вкусные подарки!",
		},
	],
]);

const promo = (lang: "en" | "ru") => [
	{
		name: (translatedIngredients.get("OPENUP") as { en: string; ru: string })[
			lang
		],
		image: "images/covers/promo/pizza-gift.jpg",
		ingredients: [
			(translatedIngredients.get("10") as { en: string; ru: string })[lang],
		],
		date: "10.04.2022",
		variants: [
			{ type: types.small[lang], price: 280 },
			{ type: types.large[lang], price: 380 },
		],
	},
	{
		name: (
			translatedIngredients.get("PIZZA1999") as { en: string; ru: string }
		)[lang],
		image: "images/covers/promo/pizza399.jpg",
		ingredients: [
			(translatedIngredients.get("1999") as { en: string; ru: string })[lang],
		],
		date: "08.04.2022",
		variants: [
			{ type: types.small[lang], price: 120 },
			{ type: types.large[lang], price: 200 },
		],
	},
	{
		name: (translatedIngredients.get("Prize") as { en: string; ru: string })[
			lang
		],
		image: "images/covers/promo/pizza-gift2.jpg",
		ingredients: [
			(translatedIngredients.get("Tasty") as { en: string; ru: string })[lang],
		],
		date: "08.04.2022",
		variants: [
			{ type: types.small[lang], price: 210 },
			{ type: types.large[lang], price: 390 },
		],
	},
	{
		name: (translatedIngredients.get("Souvenir") as { en: string; ru: string })[
			lang
		],
		image: "images/covers/promo/souvenir.jpg",
		ingredients: [
			(translatedIngredients.get("Tasty") as { en: string; ru: string })[lang],
		],
		date: "07.04.2022",
		variants: [
			{ type: types.small[lang], price: 200 },
			{ type: types.large[lang], price: 380 },
		],
	},
	{
		name: (
			translatedIngredients.get("Invite") as {
				en: string;
				ru: string;
			}
		)[lang],
		image: "images/covers/promo/friend.jpg",
		ingredients: [
			(translatedIngredients.get("Tasty") as { en: string; ru: string })[lang],
		],
		date: "07.04.2022",
		variants: [
			{ type: types.small[lang], price: 210 },
			{ type: types.large[lang], price: 390 },
		],
	},
	{
		name: (translatedIngredients.get("Birthday") as { en: string; ru: string })[
			lang
		],
		image: "images/covers/promo/birth.jpg",
		ingredients: [
			(translatedIngredients.get("Tasty") as { en: string; ru: string })[lang],
		],
		date: "04.04.2022",
		variants: [
			{ type: types.small[lang], price: 150 },
			{ type: types.large[lang], price: 270 },
		],
	},
	{
		name: (
			translatedIngredients.get("FirstOrder") as { en: string; ru: string }
		)[lang],
		image: "images/covers/promo/2000first.jpg",
		ingredients: [
			(translatedIngredients.get("Tasty") as { en: string; ru: string })[lang],
		],
		date: "04.04.2022",
		variants: [
			{ type: types.small[lang], price: 60 },
			{ type: types.large[lang], price: 100 },
		],
	},
	{
		name: (
			translatedIngredients.get("FIRST1") as {
				en: string;
				ru: string;
			}
		)[lang],
		image: "images/covers/promo/30percent.jpg",
		ingredients: [
			(translatedIngredients.get("30") as { en: string; ru: string })[lang],
		],
		date: "03.04.2022",
		variants: [
			{ type: types.small[lang], price: 70 },
			{ type: types.large[lang], price: 120 },
		],
	},
	{
		name: (translatedIngredients.get("MakeSet") as { en: string; ru: string })[
			lang
		],
		image: "images/covers/promo/welcome.jpg",
		ingredients: [
			(translatedIngredients.get("Tasty") as { en: string; ru: string })[lang],
		],
		date: "03.04.2022",
		variants: [
			{ type: types.small[lang], price: 150 },
			{ type: types.large[lang], price: 250 },
		],
	},
];

export default promo;
