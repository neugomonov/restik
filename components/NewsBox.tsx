import { ArrowForwardIcon } from "@chakra-ui/icons";
import {
	Box,
	Button,
	Heading,
	IconButton,
	Image,
	Stack,
	Tag,
	Text,
	useColorMode,
} from "@chakra-ui/react";
import { motion } from "framer-motion";
import useTranslation from "next-translate/useTranslation";
import { useRouter } from "next/router";
import { BiNews } from "react-icons/bi";
import { useRecoilState } from "recoil";
import info from "../lib/info";
import news from "../lib/news";
import { _blur } from "../lib/recoil-atoms";

export default function NewsBox() {
	const router = useRouter();
	const { t, lang } = useTranslation("common");
	const { colorMode } = useColorMode();
	const handleClick = (route: string) => {
		return async () => {
			await router.push(route, route);
		};
	};
	const [blurMode] = useRecoilState(_blur);

	return (
		<Box
			transition=".3s ease"
			borderWidth="1px"
			borderRadius="lg"
			padding="1rem"
			width={{ base: "100%", xl: "100%" }}
			boxShadow="rgba(255, 255, 255, 0.1) 0px 1px 1px 0px inset, rgba(50, 50, 93, 0.25) 0px 50px 100px -20px, rgba(0, 0, 0, 0.3) 0px 30px 60px -30px"
			backgroundColor={
				colorMode === "dark"
					? "rgba(6, 8, 13, 0.75)"
					: "rgba(255, 255, 255, 0.75)"
			}
			position="sticky"
			top="100%"
			backdropFilter={blurMode.blur ? "auto" : "none"}
			backdropBlur="20px"
		>
			<div
				style={{
					display: "flex",
					alignItems: "center",
					justifyContent: "space-between",
				}}
			>
				{info.isDevelopment && (
					<Tag
						as={motion.div}
						cursor="pointer"
						drag
						dragConstraints={{ top: 0, left: 0, right: 0, bottom: 0 }}
						whileDrag={{ scale: 1.2, rotate: 10 }}
						dragTransition={{ bounceStiffness: 1399, bounceDamping: 10 }}
						whileTap={{
							scale: 0.9,
						}}
						whileHover={{
							scale: 1.2,
							transition: { type: "spring", bounce: 0.8, duration: 1 },
						}}
						textTransform="uppercase"
						colorScheme="orange"
						variant="solid"
						mb="1rem"
					>
						{info.news[lang as "en" | "ru"]}
					</Tag>
				)}
				<Box
					as={motion.div}
					drag
					dragConstraints={{ top: 0, left: 0, right: 0, bottom: 0 }}
					whileDrag={{ scale: 1.2, rotate: -45 }}
					dragTransition={{ bounceStiffness: 1399, bounceDamping: 10 }}
					whileTap={{
						scale: 0.9,
					}}
					whileHover={{
						scale: 1.2,
						transition: { type: "spring", bounce: 0.8, duration: 1 },
					}}
				>
					<IconButton aria-label="News" icon={<BiNews />} />
				</Box>
			</div>
			{news(lang as "en" | "ru")
				.map((item) => (
					<Box padding="1rem">
						<Stack spacing={3}>
							<Image
								src={`/${item.image}`}
								alt={`${t("photoOf")} ${item.name}`}
								draggable={false}
								loading="lazy"
								decoding="async"
								width="auto"
								height={150}
								objectFit="cover"
								borderRadius="md"
							/>
							<Text colorScheme={"gray"}>{item.date}</Text>

							<Heading mr="1%">{item.name}</Heading>
							<Text colorScheme={"gray"}>{item.ingredients.join(", ")}</Text>
							<Button
								rightIcon={<ArrowForwardIcon />}
								colorScheme="orange"
								variant="outline"
								data-testid="button"
								onClick={handleClick("/news")}
							>
								{info.news[lang as "en" | "ru"] ?? t("sampleButton")}
							</Button>
						</Stack>
					</Box>
				))
				.at(Math.random() * (-1 - 9 + 1) + 9)}
		</Box>
	);
}
