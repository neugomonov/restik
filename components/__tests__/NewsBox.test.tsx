import { render, screen, cleanup } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import NewsBox from "../NewsBox";
import renderer from "react-test-renderer";
import { RecoilRoot } from "recoil";
import { _cart, CartState } from "../../lib/recoil-atoms";
import StateSaver from "../../components/state-saver";

afterEach(cleanup);

it("renders without crashing", () => {
	render(
		<RecoilRoot initializeState={({ set }) => {}}>
			<StateSaver>
				<NewsBox />
			</StateSaver>
		</RecoilRoot>
	);
});
it("renders component correctly", () => {
	const { getByTestId } = render(
		<RecoilRoot initializeState={({ set }) => {}}>
			<StateSaver>
				<NewsBox />
			</StateSaver>
		</RecoilRoot>
	);
	expect(getByTestId("button")).toHaveTextContent("Новости");
});
it("allows me to go to news", () => {
	const { getByTestId } = render(
		<RecoilRoot initializeState={({ set }) => {}}>
			<StateSaver>
				<NewsBox />
			</StateSaver>
		</RecoilRoot>
	);
	userEvent.click(screen.getByTestId("button"));
	expect(getByTestId("button")).toHaveTextContent("Новости");
});
it("matches snapshot", () => {
	const tree = renderer
		.create(
			<RecoilRoot initializeState={({ set }) => {}}>
				<StateSaver>
					<NewsBox />
				</StateSaver>
			</RecoilRoot>
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
