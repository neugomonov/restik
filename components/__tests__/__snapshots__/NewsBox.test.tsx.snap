// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="css-16ng1kq"
>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "space-between",
      }
    }
  >
    <span
      className="css-124qy5l"
    >
      –ù–æ–≤–æ—Å—Ç–∏
       
    </span>
    <button
      aria-label="News"
      className="chakra-button css-198bfjk"
      type="button"
    >
      <svg
        aria-hidden={true}
        fill="currentColor"
        focusable={false}
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19.875 3H4.125C2.953 3 2 3.897 2 5v14c0 1.103.953 2 2.125 2h15.75C21.047 21 22 20.103 22 19V5c0-1.103-.953-2-2.125-2zm0 16H4.125c-.057 0-.096-.016-.113-.016-.007 0-.011.002-.012.008L3.988 5.046c.007-.01.052-.046.137-.046h15.75c.079.001.122.028.125.008l.012 13.946c-.007.01-.052.046-.137.046z"
        />
        <path
          d="M6 7h6v6H6zm7 8H6v2h12v-2h-4zm1-4h4v2h-4zm0-4h4v2h-4z"
        />
      </svg>
    </button>
  </div>
  <div
    className="css-q5fqw0"
  >
    <div
      className="chakra-stack css-15duo8n"
    >
      <img
        className="chakra-image css-145mpb9"
        decoding="async"
        draggable={false}
        loading="lazy"
        src="/images/covers/news/openup.jpg"
      />
      <p
        className="chakra-text css-0"
      >
        03.04.2022
      </p>
      <h2
        className="chakra-heading css-1xuv852"
      >
        –ú—ã –æ—Ç–∫—Ä—ã–ª–∏—Å—å! üéâ 
      </h2>
      <p
        className="chakra-text css-0"
      >
        –ó–∞–∫–∞–∑—ã–≤–∞–π—Ç–µ –¥–æ–º–æ–π, –µ—à—å—Ç–µ —É –Ω–∞—Å, —Å–ª–µ–¥–∏—Ç–µ –∑–∞ –Ω–∞—à–∏–º–∏ –Ω–æ–≤–æ—Å—Ç—è–º–∏, –±—É–¥–µ—Ç –º–Ω–æ–≥–æ –∏–Ω—Ç–µ—Ä–µ—Å–Ω–æ–≥–æ. –ò –≤–∫—É—Å–Ω–æ–≥–æ! üçï
      </p>
      <button
        className="chakra-button css-5c2176"
        data-testid="button"
        onClick={[Function]}
        type="button"
      >
        –ù–æ–≤–æ—Å—Ç–∏
        <span
          className="chakra-button__icon css-1hzyiq5"
        >
          <svg
            aria-hidden={true}
            className="chakra-icon css-onkibi"
            focusable={false}
            viewBox="0 0 24 24"
          >
            <path
              d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
              fill="currentColor"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</div>
`;
